internal/infrastructure/storage/db/badger/unspent_repository_impl.go

```diff
@@ -259,129 +257,165 @@ func (u unspentRepositoryImpl) getBalance(
 func (u unspentRepositoryImpl) spendUnspents(
 	ctx context.Context,
 	unspentKeys []domain.UnspentKey,
-) error {
+) (int, error) {
+	count := 0
 	for _, key := range unspentKeys {
-		if err := u.spendUnspent(ctx, key); err != nil {
-			return err
+		done, err := u.spendUnspent(ctx, key)
+		if err != nil {
+			return -1, err
+		}
+		if done {
+			count++
 		}
 	}
-	return nil
+	return count, nil
 }
 
 func (u unspentRepositoryImpl) spendUnspent(
 	ctx context.Context,
 	key domain.UnspentKey,
-) error {
+) (bool, error) {
 	unspent, err := u.getUnspent(ctx, key)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	if unspent == nil {
-		return nil
+		return false, nil
 	}
 
 	unspent.Spend()
 	unspent.Unlock() // prevent conflict, locks not stored under unspent prefix
 
-	return u.updateUnspent(ctx, key, *unspent)
+	if err := u.updateUnspent(ctx, key, *unspent); err != nil {
+		return false, err
+	}
+
+	return true, nil
 }
 
 func (u unspentRepositoryImpl) confirmUnspents(
 	ctx context.Context,
 	unspentKeys []domain.UnspentKey,
-) error {
+) (int, error) {
+	count := 0
 	for _, key := range unspentKeys {
-		if err := u.confirmUnspent(ctx, key); err != nil {
-			return err
+		done, err := u.confirmUnspent(ctx, key)
+		if err != nil {
+			return -1, err
+		}
+		if done {
+			count++
 		}
 	}
-	return nil
+	return count, nil
 }
 
 func (u unspentRepositoryImpl) confirmUnspent(
 	ctx context.Context,
 	key domain.UnspentKey,
-) error {
+) (bool, error) {
 	unspent, err := u.getUnspent(ctx, key)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	if unspent == nil {
-		return nil
+		return false, nil
 	}
 
 	unspent.Confirm()
 	unspent.Unlock() // prevent conflict, locks not stored under unspent prefix
 
-	return u.updateUnspent(ctx, key, *unspent)
+	if err := u.updateUnspent(ctx, key, *unspent); err != nil {
+		return false, err
+	}
+
+	return true, nil
 }
 
 func (u unspentRepositoryImpl) lockUnspents(
 	ctx context.Context,
 	unspentKeys []domain.UnspentKey,
 	tradeID uuid.UUID,
-) error {
+) (int, error) {
+	count := 0
 	for _, key := range unspentKeys {
-		if err := u.lockUnspent(ctx, key, tradeID); err != nil {
-			return err
+		done, err := u.lockUnspent(ctx, key, tradeID)
+		if err != nil {
+			return -1, err
+		}
+		if done {
+			count++
 		}
 	}
-	return nil
+	return count, nil
 }
 
 func (u unspentRepositoryImpl) lockUnspent(
 	ctx context.Context,
 	key domain.UnspentKey,
 	tradeID uuid.UUID,
-) error {
+) (bool, error) {
 	unspent, err := u.getUnspent(ctx, key)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	if unspent == nil {
-		return nil
+		return false, nil
 	}
 
 	if unspent.IsLocked() {
-		return nil
+		return false, nil
+	}
+
+	if err := u.insertLock(ctx, key, tradeID); err != nil {
+		return false, err
 	}
 
-	return u.insertLock(ctx, key, tradeID)
+	return true, nil
 }
 
 func (u unspentRepositoryImpl) unlockUnspents(
```

Comment:
It looks liek here we are dropping TTL therefore we could have used `badgerhold` abstractions. But maybe it's ok like this

Reply:
TTLs are not ok anymore because we need to need to update the lock and spent status together, otherwise there could be the chance that some actually spent utxos might be used for some trades just because the spent status is updated AFTER the trade proposal.

Reply:
let's open a ticket for that.